From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 20 May 2023 13:21:34 +0200
Subject: [PATCH] TenseiMC Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 1ef4cf3b19d58565e052829358eb5902f070ae02..2e12ad1f7424b594ffb99f385c33a0e0e507f90e 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2007,6 +2007,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // TenseiMC start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getTenseiConfig()  {
+            throw new UnsupportedOperationException("Not supported yet");
+        }
+        // TenseiMC end
+
         /**
          * Sends the component to the player
          *
