From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 20 May 2023 13:21:34 +0200
Subject: [PATCH] TenseiMC Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 5e427d5c5ce58927eb87b4f4f2466a27e70bc7c1..fb29becaecf0b9e3bd92b22c0a66daa4083fa687 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1994,6 +1994,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // TenseiMC start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getTenseiConfig()  {
+            throw new UnsupportedOperationException("Not supported yet");
+        }
+        // TenseiMC end
+
         /**
          * Sends the component to the player
          *
