From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 20 May 2023 13:21:34 +0200
Subject: [PATCH] TenseiMC Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 85b169c04f44431363d4e14d4857140f160ceace..d56ce86527179537e7c92ba8d8fbbdd68b1b662c 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2190,6 +2190,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // TenseiMC start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getTenseiConfig()  {
+            throw new UnsupportedOperationException("Not supported yet");
+        }
+        // TenseiMC end
+
         /**
          * Sends the component to the player
          *
