From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 20 May 2023 13:21:34 +0200
Subject: [PATCH] TenseiMC Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index f77553fc70bc1f3f18af50e53c0041bbaaebab9a..c36ab634529c5a426bdffc7ec52bd88ce57ac46c 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2081,6 +2081,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // TenseiMC start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getTenseiConfig()  {
+            throw new UnsupportedOperationException("Not supported yet");
+        }
+        // TenseiMC end
+
         /**
          * Sends the component to the player
          *
