From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 10 Jun 2023 01:03:50 +0200
Subject: [PATCH] Method Update Chunk Client


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 0423483f11d45160d501fc821ada2b8104188628..2fa534552dd18d71cb4a6ef0d34afac90219cf1f 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2124,6 +2124,15 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     @org.jetbrains.annotations.ApiStatus.Experimental
     public void setOwnWorldBorder(@NotNull Location centerBorder, @NotNull fr.euphilia.utils.BorderColor colorBorder, double borderSize, int warningBlocks, int warningTime);
+
+    /**
+     * Update the chunk on the player's client.
+     *
+     * @param chunkX X coordinates of the chunk.
+     * @param chunkZ Z coordinates of the chunk.
+     */
+    @org.jetbrains.annotations.ApiStatus.Experimental
+    public void refreshPlayerChunk(int chunkX, int chunkZ);
     // TenseiMC end
 
     /**
