From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Tue, 1 Aug 2023 12:41:32 +0200
Subject: [PATCH] Purpur - EnderChest 1 to 6 Rows


diff --git a/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java b/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java
index c704b742c627589de92322707ef46a21ed9b8100..a018ab14d641883e091b6a0e2a2a475549378869 100644
--- a/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java
+++ b/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java
@@ -16,6 +16,14 @@ public class TenseiMCPermissions {
 
         // Add Permissions here
 
+        // Enderchest Permissions Purpur
+        DefaultPermissions.registerPermission("purpur." + "enderchest.rows.six", "Gives the user six rows of enderchest space.", PermissionDefault.FALSE, tenseimc);
+        DefaultPermissions.registerPermission("purpur." + "enderchest.rows.five", "Gives the user five rows of enderchest space.", PermissionDefault.FALSE, tenseimc);
+        DefaultPermissions.registerPermission("purpur." + "enderchest.rows.four", "Gives the user four rows of enderchest space.", PermissionDefault.FALSE, tenseimc);
+        DefaultPermissions.registerPermission("purpur." + "enderchest.rows.three", "Gives the user three rows of enderchest space.", PermissionDefault.FALSE, tenseimc);
+        DefaultPermissions.registerPermission("purpur." + "enderchest.rows.two", "Gives the user two rows of enderchest space.", PermissionDefault.FALSE, tenseimc);
+        DefaultPermissions.registerPermission("purpur." + "enderchest.rows.one", "Gives the user one row of enderchest space.", PermissionDefault.FALSE, tenseimc);
+
         tenseimc.recalculatePermissibles();
 
         return tenseimc;
