From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Tue, 1 Aug 2023 12:43:38 +0200
Subject: [PATCH] Allow player join full server by permission


diff --git a/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java b/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java
index a018ab14d641883e091b6a0e2a2a475549378869..20a65e2b8f4e9ea916ef0fbf9d90e4f7d091cf84 100644
--- a/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java
+++ b/src/main/java/org/bukkit/util/permissions/TenseiMCPermissions.java
@@ -24,6 +24,9 @@ public class TenseiMCPermissions {
         DefaultPermissions.registerPermission("purpur." + "enderchest.rows.two", "Gives the user two rows of enderchest space.", PermissionDefault.FALSE, tenseimc);
         DefaultPermissions.registerPermission("purpur." + "enderchest.rows.one", "Gives the user one row of enderchest space.", PermissionDefault.FALSE, tenseimc);
 
+        // Other
+        DefaultPermissions.registerPermission("purpur." + "joinfullserver", "Allows the user to join a full server.", PermissionDefault.OP, tenseimc);
+
         tenseimc.recalculatePermissibles();
 
         return tenseimc;
