From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <blake.galbreath@gmail.com>
Date: Tue, 23 May 2023 07:15:04 +0200
Subject: [PATCH] Purpur - Barrels 6 rows


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
index 9265ca7f76fc3ace51ac69554ea3853ed2042f41..bddda44bdab13d455d7a3c415d8b4e8429fa1eff 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
@@ -125,6 +125,7 @@ public class TenseiConfig {
 
     public static boolean enderChestSixRows = false;
     public static boolean enderChestPermissionRows = false;
+    public static int barrelRows = 3;
     private static void blockSettings() {
         if (version < 2) {
             enderChestSixRows = getBoolean("settings.blocks.ender_chest.six-rows", enderChestSixRows);
@@ -132,12 +133,16 @@ public class TenseiConfig {
         if (version < 2) {
             enderChestPermissionRows = getBoolean("settings.blocks.ender_chest.use-permissions-for-rows", enderChestPermissionRows);
         }
+        if (version < 2) {
+            barrelRows = getInt("settings.blocks.barrel.rows", barrelRows);
+        }
     }
 
     private static void updateConfigVersion() {
         if (version >= 2) {
             set("settings.blocks.ender_chest.six-rows", null);
             set("settings.blocks.ender_chest.use-permissions-for-rows", null);
+            set("settings.blocks.barrel.rows", null);
         }
     }
 
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BarrelBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
index 416aa989ebb18a8741cc9d605a1180ab830f6643..89c637babe8e0b47ffd86e3a0c239f1691347774 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
@@ -67,7 +67,7 @@ public class BarrelBlockEntity extends RandomizableContainerBlockEntity {
 
     public BarrelBlockEntity(BlockPos pos, BlockState state) {
         super(BlockEntityType.BARREL, pos, state);
-        this.items = NonNullList.withSize(27, ItemStack.EMPTY);
+        this.items = NonNullList.withSize(org.purpurmc.purpur.PurpurConfig.barrelRows*9, ItemStack.EMPTY); // TenseiMC start
         this.openersCounter = new ContainerOpenersCounter() {
             @Override
             protected void onOpen(Level world, BlockPos pos, BlockState state) {
@@ -118,7 +118,7 @@ public class BarrelBlockEntity extends RandomizableContainerBlockEntity {
 
     @Override
     public int getContainerSize() {
-        return 27;
+        return org.purpurmc.purpur.PurpurConfig.barrelRows*9; // TenseiMC start
     }
 
     @Override
@@ -138,7 +138,16 @@ public class BarrelBlockEntity extends RandomizableContainerBlockEntity {
 
     @Override
     protected AbstractContainerMenu createMenu(int syncId, Inventory playerInventory) {
-        return ChestMenu.threeRows(syncId, playerInventory, this);
+        // TenseiMC Start
+        return switch (org.purpurmc.purpur.PurpurConfig.barrelRows) {
+            case 6 -> ChestMenu.sixRows(syncId, playerInventory, this);
+            case 5 -> ChestMenu.fiveRows(syncId, playerInventory, this);
+            case 4 -> ChestMenu.fourRows(syncId, playerInventory, this);
+            case 2 -> ChestMenu.twoRows(syncId, playerInventory, this);
+            case 1 -> ChestMenu.oneRows(syncId, playerInventory, this);
+            default -> ChestMenu.threeRows(syncId, playerInventory, this);
+        };
+        // TenseiMC end
     }
 
     @Override
diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
index 2730e8fd695a9bad0e991fcb98f6127cc895557f..e466f35911943de42f3ca808ce31b0254a8af4fe 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftContainer.java
@@ -188,7 +188,16 @@ public class CraftContainer extends AbstractContainerMenu {
                 break;
                 // TenseiMC end
             case BARREL:
-                this.delegate = new ChestMenu(MenuType.GENERIC_9x3, windowId, bottom, top, top.getContainerSize() / 9);
+                // TenseiMC start
+                this.delegate = new ChestMenu(switch (org.purpurmc.purpur.PurpurConfig.barrelRows) {
+                    case 6 -> MenuType.GENERIC_9x6;
+                    case 5 -> MenuType.GENERIC_9x5;
+                    case 4 -> MenuType.GENERIC_9x4;
+                    case 2 -> MenuType.GENERIC_9x2;
+                    case 1 -> MenuType.GENERIC_9x1;
+                    default -> MenuType.GENERIC_9x3;
+                }, windowId, bottom, top, top.getContainerSize() / 9);
+                // TenseiMC end
                 break;
             case DISPENSER:
             case DROPPER:
diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index 840aef59f276b25af8a28c99d94906fd0199c232..9841b60b5806a579191470882330f48e850f9fdd 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -173,9 +173,20 @@ public class PurpurConfig {
         return builder.build();
     }
 
+    public static int barrelRows = 3;
     public static boolean enderChestSixRows = false;
     public static boolean enderChestPermissionRows = false;
     private static void blockSettings() {
+        if (fr.euphilia.tenseimc.TenseiConfig.version < 2) {
+            barrelRows = fr.euphilia.tenseimc.TenseiConfig.barrelRows;
+            set("settings.blocks.barrel.rows", barrelRows);
+        }
+        barrelRows = getInt("settings.blocks.barrel.rows", barrelRows);
+        if (barrelRows < 1 || barrelRows > 6) {
+            Bukkit.getLogger().severe("settings.blocks.barrel.rows must be 1-6, resetting to default");
+            barrelRows = 3;
+        }
+        org.bukkit.event.inventory.InventoryType.BARREL.setDefaultSize(barrelRows*9);
         if (fr.euphilia.tenseimc.TenseiConfig.version < 2) {
             enderChestSixRows = fr.euphilia.tenseimc.TenseiConfig.enderChestSixRows;
             set("settings.blocks.ender_chest.six-rows", enderChestSixRows);
