From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sun, 4 Jun 2023 10:52:28 +0200
Subject: [PATCH] Beacon Activation Range Configurable


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
index 181bae070ce354b1e199e644a1089ac758070593..9c33a3cf407d0a014fbc85053a4eafd30d310ff2 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
@@ -64,6 +64,19 @@ public class TenseiWorldConfig {
         return value.isEmpty() ? fallback : value;
     }
 
+    public int beaconLevelOne = 20;
+    public int beaconLevelTwo = 30;
+    public int beaconLevelThree = 40;
+    public int beaconLevelFour = 50;
+    private void beaconSettings() {
+        if (TenseiConfig.version < 2) {
+            beaconLevelOne = getInt("blocks.beacon.effect-range.level-1", beaconLevelOne);
+            beaconLevelTwo = getInt("blocks.beacon.effect-range.level-2", beaconLevelTwo);
+            beaconLevelThree = getInt("blocks.beacon.effect-range.level-3", beaconLevelThree);
+            beaconLevelFour = getInt("blocks.beacon.effect-range.level-4", beaconLevelFour);
+        }
+    }
+
     public boolean disableObserverClocks = false;
     public int pistonBlockPushLimit = 12;
     private void redstoneSettings() {
@@ -78,6 +91,7 @@ public class TenseiWorldConfig {
         if (TenseiConfig.version >= 2) {
             set("blocks.observer.disable-clock", null);
             set("blocks.piston.block-push-limit", null);
+            set("blocks.beacon.effect-range", null);
         }
     }
 }
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
index 9c3a271f98e723f1d8bf3badd3fca7a19fdc6d13..3ff48acdf99e9aca2109f19cd893aece56031f05 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
@@ -88,6 +88,17 @@ public class BeaconBlockEntity extends BlockEntity implements MenuProvider, Name
 
     public double getEffectRange() {
         if (this.effectRange < 0) {
+            // TenseiMC start
+            if (this.level != null) {
+                return switch (this.levels) {
+                    case 1 -> this.level.purpurConfig.beaconLevelOne;
+                    case 2 -> this.level.purpurConfig.beaconLevelTwo;
+                    case 3 -> this.level.purpurConfig.beaconLevelThree;
+                    case 4 -> this.level.purpurConfig.beaconLevelFour;
+                    default -> this.levels * 10 + 10;
+                };
+            }
+            // TenseiMC end
             return this.levels * 10 + 10;
         } else {
             return effectRange;
diff --git a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
index fab41119d81589185731bfdaffcf3d56625a0867..5a3aac37763ac3bdd4044f262d4acda009cad51d 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -111,4 +111,25 @@ public class PurpurWorldConfig {
         }
         pistonBlockPushLimit = getInt("blocks.piston.block-push-limit", pistonBlockPushLimit);
     }
+
+    public int beaconLevelOne = 20;
+    public int beaconLevelTwo = 30;
+    public int beaconLevelThree = 40;
+    public int beaconLevelFour = 50;
+    private void beaconSettings() {
+        if (TenseiConfig.version < 2) {
+            beaconLevelOne = this.tenseiWorldConfig.beaconLevelOne;
+            beaconLevelTwo = this.tenseiWorldConfig.beaconLevelTwo;
+            beaconLevelThree = this.tenseiWorldConfig.beaconLevelThree;
+            beaconLevelFour = this.tenseiWorldConfig.beaconLevelFour;
+            set("blocks.beacon.effect-range.level-1", beaconLevelOne);
+            set("blocks.beacon.effect-range.level-2", beaconLevelTwo);
+            set("blocks.beacon.effect-range.level-3", beaconLevelThree);
+            set("blocks.beacon.effect-range.level-4", beaconLevelFour);
+        }
+        beaconLevelOne = getInt("blocks.beacon.effect-range.level-1", beaconLevelOne);
+        beaconLevelTwo = getInt("blocks.beacon.effect-range.level-2", beaconLevelTwo);
+        beaconLevelThree = getInt("blocks.beacon.effect-range.level-3", beaconLevelThree);
+        beaconLevelFour = getInt("blocks.beacon.effect-range.level-4", beaconLevelFour);
+    }
 }
\ No newline at end of file
