From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 10 Jun 2023 00:58:38 +0200
Subject: [PATCH] Method Custom WorldBorder Player


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index bc59a9d2f965c8e8638a2eeb3fcf661ed81f1727..467df1d20682be31459743350f0ec64b4e2ac5cf 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1112,6 +1112,26 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         connection.send(new ClientboundSetBorderWarningDistancePacket(newWorldBorder));
     }
 
+    // TenseiMC start
+    @org.jetbrains.annotations.ApiStatus.Experimental
+    @Override
+    public void setOwnWorldBorder(@NotNull Location centerBorder, @NotNull fr.euphilia.utils.BorderColor colorBorder, double borderSize, int warningBlocks, int warningTime) {
+        final net.minecraft.world.level.border.WorldBorder worldBorderPlayer = new net.minecraft.world.level.border.WorldBorder();
+        worldBorderPlayer.world = ((org.bukkit.craftbukkit.CraftWorld) centerBorder.getWorld()).getHandle();
+        worldBorderPlayer.setCenter(centerBorder.getBlockX(), centerBorder.getBlockZ());
+        worldBorderPlayer.setSize(borderSize);
+        worldBorderPlayer.setWarningBlocks(warningBlocks);
+        worldBorderPlayer.setWarningTime(warningTime);
+        switch (colorBorder) {
+            case RED -> worldBorderPlayer.lerpSizeBetween(borderSize, borderSize -1.0, 20000000L);
+            case GREEN -> worldBorderPlayer.lerpSizeBetween(borderSize - 0.1, borderSize, 20000000L);
+            case BLUE -> worldBorderPlayer.lerpSizeBetween(borderSize, 0L, 0L);
+        }
+        final net.minecraft.network.protocol.game.ClientboundInitializeBorderPacket updateWorldBorderPacket = new net.minecraft.network.protocol.game.ClientboundInitializeBorderPacket(worldBorderPlayer);
+        this.getHandle().connection.send(updateWorldBorderPacket);
+    }
+    // TenseiMC end
+
     private BorderChangeListener createWorldBorderListener() {
         return new BorderChangeListener() {
             @Override
