From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 10 Jun 2023 01:03:59 +0200
Subject: [PATCH] Method Update Chunk Client


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 2ba0555a222d5a85db9eac806c8ce35572fb2971..c39733f4755f6354ea2f32323abb5f4e543d5bf7 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1120,6 +1120,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         final net.minecraft.network.protocol.game.ClientboundInitializeBorderPacket updateWorldBorderPacket = new net.minecraft.network.protocol.game.ClientboundInitializeBorderPacket(worldBorderPlayer);
         this.getHandle().connection.send(updateWorldBorderPacket);
     }
+
+    @org.jetbrains.annotations.ApiStatus.Experimental
+    @Override
+    public void refreshPlayerChunk(int chunkX, int chunkZ) {
+        final net.minecraft.world.level.chunk.LevelChunk levelChunk = this.getHandle().level().getChunk(chunkX, chunkZ);
+        final net.minecraft.network.protocol.game.ClientboundLevelChunkWithLightPacket refresh = new net.minecraft.network.protocol.game.ClientboundLevelChunkWithLightPacket(levelChunk, levelChunk.getLevel().getLightEngine(), null, null);
+        this.getHandle().connection.send(refresh);
+    }
     // TenseiMC end
 
     private BorderChangeListener createWorldBorderListener() {
