From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sun, 11 Jun 2023 12:33:43 +0200
Subject: [PATCH] Add config change multiplier critical damage value


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
index c6de83a7f275d218c61925531ae825b4e1a98202..504ee522b9be18dfbe9a8690c4a06e20e283c709 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
@@ -81,4 +81,9 @@ public class TenseiWorldConfig {
         beaconLevelThree = getInt("blocks.beacon.effect-range.level-3", beaconLevelThree);
         beaconLevelFour = getInt("blocks.beacon.effect-range.level-4", beaconLevelFour);
     }
+
+    public double playerCriticalDamageMultiplier = 1.5D;
+    private void playerSettings() {
+        playerCriticalDamageMultiplier = getDouble("gameplay-mechanics.player.critical-damage-multiplier", playerCriticalDamageMultiplier);
+    }
 }
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 97efc245e0e1dcbd5ba1c602c0a67dcf403cb7a5..c9b26438fa496b52bf329cc2d00c5bd046f4e4cf 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -1289,7 +1289,7 @@ public abstract class Player extends LivingEntity {
                     flag2 = flag2 && !this.level().paperConfig().entities.behavior.disablePlayerCrits; // Paper
                     flag2 = flag2 && !this.isSprinting();
                     if (flag2) {
-                        f *= 1.5F;
+                        f *= this.level().tenseiWorldConfig.playerCriticalDamageMultiplier; // TenseiMC
                     }
 
                     f += f1;
