From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Wed, 14 Jun 2023 18:39:27 +0200
Subject: [PATCH] "Bee can work when raining or at night"


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
index efd8bb64030dbdb4655cd01d11cfb41d4de615d4..2e6108681fa25c2ad61f3bdcccac11d11fb37898 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
@@ -86,6 +86,15 @@ public class TenseiWorldConfig {
         }
     }
 
+    public boolean beeCanWorkAtNight = false;
+    public boolean beeCanWorkInRain = false;
+    private void mobsSettings() {
+        if (TenseiConfig.version < 2) {
+            beeCanWorkAtNight = getBoolean("mobs.bee.can-work-at-night", beeCanWorkAtNight);
+            beeCanWorkInRain = getBoolean("mobs.bee.can-work-in-rain", beeCanWorkInRain);
+        }
+    }
+
     public double playerCriticalDamageMultiplier = 1.5D;
     private void playerSettings() {
         if (TenseiConfig.version < 2) {
@@ -110,6 +119,8 @@ public class TenseiWorldConfig {
             set("blocks.beacon.effect-range", null);
             set("gameplay-mechanics.player.critical-damage-multiplier", null);
             set("blocks.chest.open-with-solid-block-on-top", null);
+            set("mobs.bee.can-work-at-night", null);
+            set("mobs.bee.can-work-in-rain", null);
         }
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
index 25fd3aea2d66503018637781ccb8b5690c02e57b..12e01b572ddfaa5284433ffb6df3fa4df4ad7fa1 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
@@ -348,7 +348,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
 
     boolean wantsToEnterHive() {
         if (this.stayOutOfHiveCountdown <= 0 && !this.beePollinateGoal.isPollinating() && !this.hasStung() && this.getTarget() == null) {
-            boolean flag = this.isTiredOfLookingForNectar() || this.level().isRaining() || this.level().isNight() || this.hasNectar();
+            boolean flag = this.isTiredOfLookingForNectar() || (this.level().isRaining() && !this.level().purpurConfig.beeCanWorkInRain) || (this.level().isNight() && !this.level().purpurConfig.beeCanWorkAtNight) || this.hasNectar(); // TenseiMC
 
             return flag && !this.isHiveNearFire();
         } else {
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
index 41c9f074203915c31c1ae7a160ce509c13383f84..f0dde6d99fa0035ffec89701691e7f18169067dd 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -203,7 +203,7 @@ public class BeehiveBlockEntity extends BlockEntity {
     }
 
     private static boolean releaseBee(Level world, BlockPos blockposition, BlockState iblockdata, BeehiveBlockEntity.BeeData tileentitybeehive_hivebee, @Nullable List<Entity> list, BeehiveBlockEntity.BeeReleaseStatus tileentitybeehive_releasestatus, @Nullable BlockPos blockposition1, boolean force) {
-        if (!force && (world.isNight() || world.isRaining()) && tileentitybeehive_releasestatus != BeehiveBlockEntity.BeeReleaseStatus.EMERGENCY) {
+        if (!force && ((world.isNight() && !world.purpurConfig.beeCanWorkAtNight) || (world.isRaining() && !world.purpurConfig.beeCanWorkInRain)) && tileentitybeehive_releasestatus != BeehiveBlockEntity.BeeReleaseStatus.EMERGENCY) { // TenseiMC
             // CraftBukkit end
             return false;
         } else {
diff --git a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
index 0d1b9d43ce81e4931c8e5beea2b29adfd3a56a35..af9c2fbcb3da068f23ccbaa9e76b3b1a06d68724 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -133,6 +133,19 @@ public class PurpurWorldConfig {
         beaconLevelFour = getInt("blocks.beacon.effect-range.level-4", beaconLevelFour);
     }
 
+    public boolean beeCanWorkAtNight = false;
+    public boolean beeCanWorkInRain = false;
+    private void beeSettings() {
+        if (TenseiConfig.version < 2) {
+            beeCanWorkAtNight = this.tenseiWorldConfig.beeCanWorkAtNight;
+            beeCanWorkInRain = this.tenseiWorldConfig.beeCanWorkInRain;
+            set("mobs.bee.can-work-at-night", beeCanWorkAtNight);
+            set("mobs.bee.can-work-in-rain", beeCanWorkInRain);
+        }
+        beeCanWorkAtNight = getBoolean("mobs.bee.can-work-at-night", beeCanWorkAtNight);
+        beeCanWorkInRain = getBoolean("mobs.bee.can-work-in-rain", beeCanWorkInRain);
+    }
+
     public boolean chestOpenWithBlockOnTop = false;
     private void chestSettings() {
         if (TenseiConfig.version < 2) {
