From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Wed, 14 Jun 2023 18:42:54 +0200
Subject: [PATCH] Config to change max number of bees


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
index f18357d363231072bd5bcf19ad056e2642231e0a..081cbd20e0cfb78ed1678447f1da3943cbccd543 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
@@ -130,6 +130,7 @@ public class TenseiConfig {
     public static boolean enderChestSixRows = false;
     public static boolean enderChestPermissionRows = false;
     public static int barrelRows = 3;
+    public static int beeInsideBeeHive = 3;
     public static HashMap<Item, Double> itemCompostable = new HashMap<>();
     private static void blockSettings() {
         if (version < 2) {
@@ -141,6 +142,9 @@ public class TenseiConfig {
         if (version < 2) {
             barrelRows = getInt("settings.blocks.barrel.rows", barrelRows);
         }
+        if (version < 2) {
+            beeInsideBeeHive = getInt("settings.blocks.beehive.max-bees-inside", beeInsideBeeHive);
+        }
         itemCompostable.clear();
         getMap("settings.blocks.composter.compostable", net.minecraft.world.level.block.ComposterBlock.composterBlockItem()).forEach((item, value) -> {
             try {
@@ -158,6 +162,7 @@ public class TenseiConfig {
             set("settings.blocks.ender_chest.six-rows", null);
             set("settings.blocks.ender_chest.use-permissions-for-rows", null);
             set("settings.blocks.barrel.rows", null);
+            set("settings.blocks.beehive.max-bees-inside", null);
         }
     }
 
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
index f0dde6d99fa0035ffec89701691e7f18169067dd..56cd23d34b5224391d86d3c4bdcb6975bc810972 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -43,7 +43,7 @@ public class BeehiveBlockEntity extends BlockEntity {
     private final List<BeehiveBlockEntity.BeeData> stored = Lists.newArrayList();
     @Nullable
     public BlockPos savedFlowerPos;
-    public int maxBees = 3; // CraftBukkit - allow setting max amount of bees a hive can hold
+    public int maxBees = org.purpurmc.purpur.PurpurConfig.beeInsideBeeHive; // CraftBukkit - allow setting max amount of bees a hive can hold // TenseiMC
 
     public BeehiveBlockEntity(BlockPos pos, BlockState state) {
         super(BlockEntityType.BEEHIVE, pos, state);
diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index 9841b60b5806a579191470882330f48e850f9fdd..caaf029022a647f1ec6ae4120e10f98f493b09ea 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -7,6 +7,7 @@ import com.mojang.datafixers.util.Pair;
 // import net.kyori.adventure.text.minimessage.MiniMessage; // TenseiMC - not implemented
 // import net.minecraft.core.registries.BuiltInRegistries; // TenseiMC - not implemented
 // import net.minecraft.resources.ResourceLocation; // TenseiMC - not implemented
+import fr.euphilia.tenseimc.TenseiConfig;
 import net.minecraft.server.MinecraftServer;
 // import net.minecraft.world.effect.MobEffect; // TenseiMC - not implemented
 // import net.minecraft.world.effect.MobEffectInstance; // TenseiMC - not implemented
@@ -174,6 +175,7 @@ public class PurpurConfig {
     }
 
     public static int barrelRows = 3;
+    public static int beeInsideBeeHive = 3;
     public static boolean enderChestSixRows = false;
     public static boolean enderChestPermissionRows = false;
     private static void blockSettings() {
@@ -196,5 +198,10 @@ public class PurpurConfig {
         enderChestSixRows = getBoolean("settings.blocks.ender_chest.six-rows", enderChestSixRows);
         org.bukkit.event.inventory.InventoryType.ENDER_CHEST.setDefaultSize(enderChestSixRows ? 54 : 27);
         enderChestPermissionRows = getBoolean("settings.blocks.ender_chest.use-permissions-for-rows", enderChestPermissionRows);
+        if (fr.euphilia.tenseimc.TenseiConfig.version < 2) {
+            beeInsideBeeHive = fr.euphilia.tenseimc.TenseiConfig.beeInsideBeeHive;
+            set("settings.blocks.beehive.max-bees-inside", beeInsideBeeHive);
+        }
+        beeInsideBeeHive = getInt("settings.blocks.beehive.max-bees-inside", beeInsideBeeHive);
     }
 }
\ No newline at end of file
