From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Wed, 14 Jun 2023 18:42:54 +0200
Subject: [PATCH] Config to change max number of bees


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
index edc68abf1d012084a21eea59f40092bb4b37a363..7922ba7c4175572bd378b50c03725a5b02d2df3f 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiConfig.java
@@ -132,6 +132,7 @@ public class TenseiConfig {
     private static boolean enderChestPermissionRows = false;
     private static int barrelRows = 3;
     public static HashMap<Item, Double> itemCompostable = new HashMap<>();
+    public static int beeInsideBeeHive = 3;
     private static void blockSettings() {
         if (version < 2) {
             PurpurConfig.enderChestSixRows = getBoolean("settings.blocks.ender_chest.six-rows", enderChestPermissionRows);
@@ -155,5 +156,9 @@ public class TenseiConfig {
             }
         });
         net.minecraft.world.level.block.ComposterBlock.loadComposterBlock();
+        if (version < 2) {
+            beeInsideBeeHive = getInt("settings.blocks.beehive.max-bees-inside", beeInsideBeeHive);
+            set("settings.blocks.beehive.max-bees-inside", null);
+        }
     }
 }
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
index f0dde6d99fa0035ffec89701691e7f18169067dd..56cd23d34b5224391d86d3c4bdcb6975bc810972 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -43,7 +43,7 @@ public class BeehiveBlockEntity extends BlockEntity {
     private final List<BeehiveBlockEntity.BeeData> stored = Lists.newArrayList();
     @Nullable
     public BlockPos savedFlowerPos;
-    public int maxBees = 3; // CraftBukkit - allow setting max amount of bees a hive can hold
+    public int maxBees = org.purpurmc.purpur.PurpurConfig.beeInsideBeeHive; // CraftBukkit - allow setting max amount of bees a hive can hold // TenseiMC
 
     public BeehiveBlockEntity(BlockPos pos, BlockState state) {
         super(BlockEntityType.BEEHIVE, pos, state);
diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index 2565d0857a0f578ca7e10e895e8a8ac303f7daf8..aa0fda51a36ddc74d76ab97570013dffd1ad7ec7 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -7,6 +7,7 @@ import com.mojang.datafixers.util.Pair;
 // import net.kyori.adventure.text.minimessage.MiniMessage; // TenseiMC - not implemented
 // import net.minecraft.core.registries.BuiltInRegistries; // TenseiMC - not implemented
 // import net.minecraft.resources.ResourceLocation; // TenseiMC - not implemented
+import fr.euphilia.tenseimc.TenseiConfig;
 import net.minecraft.server.MinecraftServer;
 // import net.minecraft.world.effect.MobEffect; // TenseiMC - not implemented
 // import net.minecraft.world.effect.MobEffectInstance; // TenseiMC - not implemented
@@ -174,6 +175,7 @@ public class PurpurConfig {
     }
 
     public static int barrelRows = 3;
+    public static int beeInsideBeeHive = 3;
     public static boolean enderChestSixRows = false;
     public static boolean enderChestPermissionRows = false;
     private static void blockSettings() {
@@ -186,5 +188,6 @@ public class PurpurConfig {
         enderChestSixRows = getBoolean("settings.blocks.ender_chest.six-rows", enderChestSixRows);
         org.bukkit.event.inventory.InventoryType.ENDER_CHEST.setDefaultSize(enderChestSixRows ? 54 : 27);
         enderChestPermissionRows = getBoolean("settings.blocks.ender_chest.use-permissions-for-rows", enderChestPermissionRows);
+        beeInsideBeeHive = getInt("settings.blocks.beehive.max-bees-inside", beeInsideBeeHive);
     }
 }
\ No newline at end of file
