From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Sat, 17 Jun 2023 12:13:37 +0200
Subject: [PATCH] Halloween options and optimizations


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
index 2e6108681fa25c2ad61f3bdcccac11d11fb37898..243755164cdfe687410d4eb602fc8c686bf7efe1 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
@@ -1,5 +1,6 @@
 package fr.euphilia.tenseimc;
 
+import fr.euphilia.tenseimc.tasks.HalloweenTask;
 import org.bukkit.World;
 import org.bukkit.configuration.ConfigurationSection;
 import org.jetbrains.annotations.NotNull;
@@ -86,6 +87,17 @@ public class TenseiWorldConfig {
         }
     }
 
+    public boolean forceHalloweenSeason = false;
+    public float chanceHeadHalloweenOnEntity = 0.25F;
+    private void halloweenSettings() {
+        forceHalloweenSeason = getBoolean("gameplay-mechanics.halloween.force", forceHalloweenSeason);
+        chanceHeadHalloweenOnEntity = (float) getDouble("gameplay-mechanics.halloween.head-chance", chanceHeadHalloweenOnEntity);
+        if (!forceHalloweenSeason && chanceHeadHalloweenOnEntity != 0) {
+            HalloweenTask halloweenTask = new HalloweenTask();
+            halloweenTask.init(this);
+        }
+    }
+
     public boolean beeCanWorkAtNight = false;
     public boolean beeCanWorkInRain = false;
     private void mobsSettings() {
diff --git a/src/main/java/fr/euphilia/tenseimc/tasks/HalloweenTask.java b/src/main/java/fr/euphilia/tenseimc/tasks/HalloweenTask.java
new file mode 100644
index 0000000000000000000000000000000000000000..f99cfda0fa9dbe04c694a829d82b17b5a3ca6b7f
--- /dev/null
+++ b/src/main/java/fr/euphilia/tenseimc/tasks/HalloweenTask.java
@@ -0,0 +1,24 @@
+package fr.euphilia.tenseimc.tasks;
+
+import fr.euphilia.tenseimc.TenseiWorldConfig;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoField;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class HalloweenTask {
+
+    public void init(TenseiWorldConfig config) {
+        ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+
+        scheduledExecutorService.scheduleAtFixedRate(() -> {
+            LocalDate localdate = LocalDate.now();
+            int i = localdate.get(ChronoField.DAY_OF_MONTH);
+            int j = localdate.get(ChronoField.MONTH_OF_YEAR);
+
+            config.forceHalloweenSeason = (j == 10 && i == 31);
+        }, 0, 60, TimeUnit.SECONDS);
+    }
+}
diff --git a/src/main/java/net/minecraft/world/entity/ambient/Bat.java b/src/main/java/net/minecraft/world/entity/ambient/Bat.java
index 5beaa849a250ea005733250ad3edfa8382224667..671c85574ed112d7291098e7cd5e93dcd0cf7b44 100644
--- a/src/main/java/net/minecraft/world/entity/ambient/Bat.java
+++ b/src/main/java/net/minecraft/world/entity/ambient/Bat.java
@@ -227,7 +227,7 @@ public class Bat extends AmbientCreature {
             int i = world.getMaxLocalRawBrightness(pos);
             byte b0 = 4;
 
-            if (Bat.isHalloween()) {
+            if (world.getMinecraftWorld().tenseiWorldConfig.forceHalloweenSeason) { // TenseiMC
                 b0 = 7;
             } else if (random.nextBoolean()) {
                 return false;
diff --git a/src/main/java/net/minecraft/world/entity/monster/AbstractSkeleton.java b/src/main/java/net/minecraft/world/entity/monster/AbstractSkeleton.java
index b319021b22c5dceba6199ed27814b2dcf47b8d50..f3fb8a193c3858cbc3d0be61db9c1fe56000407f 100644
--- a/src/main/java/net/minecraft/world/entity/monster/AbstractSkeleton.java
+++ b/src/main/java/net/minecraft/world/entity/monster/AbstractSkeleton.java
@@ -161,11 +161,7 @@ public abstract class AbstractSkeleton extends Monster implements RangedAttackMo
         this.reassessWeaponGoal();
         this.setCanPickUpLoot(this.level().paperConfig().entities.behavior.mobsCanAlwaysPickUpLoot.skeletons || randomsource.nextFloat() < 0.55F * difficulty.getSpecialMultiplier()); // Paper
         if (this.getItemBySlot(EquipmentSlot.HEAD).isEmpty()) {
-            LocalDate localdate = LocalDate.now();
-            int i = localdate.get(ChronoField.DAY_OF_MONTH);
-            int j = localdate.get(ChronoField.MONTH_OF_YEAR);
-
-            if (j == 10 && i == 31 && randomsource.nextFloat() < 0.25F) {
+            if (this.level().tenseiWorldConfig.forceHalloweenSeason && randomsource.nextFloat() < this.level().tenseiWorldConfig.chanceHeadHalloweenOnEntity) { // TenseiMC
                 this.setItemSlot(EquipmentSlot.HEAD, new ItemStack(randomsource.nextFloat() < 0.1F ? Blocks.JACK_O_LANTERN : Blocks.CARVED_PUMPKIN));
                 this.armorDropChances[EquipmentSlot.HEAD.getIndex()] = 0.0F;
             }
diff --git a/src/main/java/net/minecraft/world/entity/monster/Zombie.java b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
index ef61663d9ba41cf02a03800c74d4ce4ccc5faec5..e2482c0632f343f1903008ea2e9a611567e7b4c5 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Zombie.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
@@ -549,11 +549,7 @@ public class Zombie extends Monster {
         }
 
         if (this.getItemBySlot(EquipmentSlot.HEAD).isEmpty()) {
-            LocalDate localdate = LocalDate.now();
-            int i = localdate.get(ChronoField.DAY_OF_MONTH);
-            int j = localdate.get(ChronoField.MONTH_OF_YEAR);
-
-            if (j == 10 && i == 31 && randomsource.nextFloat() < 0.25F) {
+            if (this.level().tenseiWorldConfig.forceHalloweenSeason && randomsource.nextFloat() < this.level().tenseiWorldConfig.chanceHeadHalloweenOnEntity) { // TenseiMC
                 this.setItemSlot(EquipmentSlot.HEAD, new ItemStack(randomsource.nextFloat() < 0.1F ? Blocks.JACK_O_LANTERN : Blocks.CARVED_PUMPKIN));
                 this.armorDropChances[EquipmentSlot.HEAD.getIndex()] = 0.0F;
             }
