From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Mon, 24 Jul 2023 07:14:57 +0200
Subject: [PATCH] Config MobEffect by world


diff --git a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
index c2b08c4088d32d4c8bfad381ad8b1530a1a0996d..c06843f9e7ec247b27d6b826a21cbbb16e57333b 100644
--- a/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
+++ b/src/main/java/fr/euphilia/tenseimc/TenseiWorldConfig.java
@@ -117,4 +117,19 @@ public class TenseiWorldConfig {
             halloweenTask.init(this);
         }
     }
+
+    public float entityMinimalHealthPoison = 1.0F;
+    public float entityHealthRegenAmount = 1.0F;
+    public float entityPoisonDegenerationAmount = 1.0F;
+    public float entityWitherDegenerationAmount = 1.0F;
+    public float humanHungerExhaustionAmount = 0.005F;
+    public float humanSaturationRegenAmount = 1.0F;
+    private void effectSettings() {
+        entityHealthRegenAmount = (float) getDouble("gameplay-mechanics.mob-effects.health-regen-amount", entityHealthRegenAmount);
+        entityMinimalHealthPoison = (float) getDouble("gameplay-mechanics.mob-effects.minimal-health-poison-amount", entityMinimalHealthPoison);
+        entityPoisonDegenerationAmount = (float) getDouble("gameplay-mechanics.mob-effects.poison-degeneration-amount", entityPoisonDegenerationAmount);
+        entityWitherDegenerationAmount = (float) getDouble("gameplay-mechanics.mob-effects.wither-degeneration-amount", entityWitherDegenerationAmount);
+        humanHungerExhaustionAmount = (float) getDouble("gameplay-mechanics.mob-effects.hunger-exhaustion-amount", humanHungerExhaustionAmount);
+        humanSaturationRegenAmount = (float) getDouble("gameplay-mechanics.mob-effects.saturation-regen-amount", humanSaturationRegenAmount);
+    }
 }
diff --git a/src/main/java/net/minecraft/world/effect/MobEffect.java b/src/main/java/net/minecraft/world/effect/MobEffect.java
index 53cc6befb752affcfec65e18365f6d369448d407..45aec04346075dfa56ed7f5e86ab7d9033c35941 100644
--- a/src/main/java/net/minecraft/world/effect/MobEffect.java
+++ b/src/main/java/net/minecraft/world/effect/MobEffect.java
@@ -57,18 +57,19 @@ public class MobEffect {
     }
 
     public void applyEffectTick(LivingEntity entity, int amplifier) {
+        fr.euphilia.tenseimc.TenseiWorldConfig tenseiWorldConfig = entity.level().tenseiWorldConfig; // TenseiMC
         if (this == MobEffects.REGENERATION) {
             if (entity.getHealth() < entity.getMaxHealth()) {
-                entity.heal(1.0F, RegainReason.MAGIC_REGEN); // CraftBukkit
+                entity.heal(tenseiWorldConfig.entityHealthRegenAmount, RegainReason.MAGIC_REGEN); // CraftBukkit // TenseiMC
             }
         } else if (this == MobEffects.POISON) {
-            if (entity.getHealth() > 1.0F) {
-                entity.hurt(entity.damageSources().poison, 1.0F);  // CraftBukkit - DamageSource.MAGIC -> CraftEventFactory.POISON
+            if (entity.getHealth() > entity.level().tenseiWorldConfig.entityMinimalHealthPoison) { // TenseiMC
+                entity.hurt(entity.damageSources().poison, tenseiWorldConfig.entityPoisonDegenerationAmount);  // CraftBukkit - DamageSource.MAGIC -> CraftEventFactory.POISON // TenseiMC
             }
         } else if (this == MobEffects.WITHER) {
-            entity.hurt(entity.damageSources().wither(), 1.0F);
+            entity.hurt(entity.damageSources().wither(), tenseiWorldConfig.entityWitherDegenerationAmount); // TenseiMC
         } else if (this == MobEffects.HUNGER && entity instanceof Player) {
-            ((Player) entity).causeFoodExhaustion(0.005F * (float) (amplifier + 1), org.bukkit.event.entity.EntityExhaustionEvent.ExhaustionReason.HUNGER_EFFECT); // CraftBukkit - EntityExhaustionEvent
+            ((Player) entity).causeFoodExhaustion(tenseiWorldConfig.humanHungerExhaustionAmount * (float) (amplifier + 1), org.bukkit.event.entity.EntityExhaustionEvent.ExhaustionReason.HUNGER_EFFECT); // CraftBukkit - EntityExhaustionEvent // TenseiMC
         } else if (this == MobEffects.SATURATION && entity instanceof Player) {
             if (!entity.level().isClientSide) {
                 // CraftBukkit start
@@ -76,7 +77,7 @@ public class MobEffect {
                 int oldFoodLevel = entityhuman.getFoodData().foodLevel;
                 org.bukkit.event.entity.FoodLevelChangeEvent event = CraftEventFactory.callFoodLevelChangeEvent(entityhuman, amplifier + 1 + oldFoodLevel);
                 if (!event.isCancelled()) {
-                    entityhuman.getFoodData().eat(event.getFoodLevel() - oldFoodLevel, 1.0F);
+                    entityhuman.getFoodData().eat(event.getFoodLevel() - oldFoodLevel, tenseiWorldConfig.humanSaturationRegenAmount); // TenseiMC
                 }
 
                 ((CraftPlayer) entityhuman.getBukkitEntity()).sendHealthUpdate();
