From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Tue, 12 Sep 2023 15:43:22 +0200
Subject: [PATCH] Reactivate command: /worldborder I haven't found any bugs


diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 4f6bd2d88219ac82c4d08c35231e5d3314f87110..798c58c74f755c1811d4d3347c9a76c3be86bb19 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -202,7 +202,7 @@ public class Commands {
         TitleCommand.register(this.dispatcher);
         //TriggerCommand.register(this.dispatcher); // Folia - region threading - TODO later
         WeatherCommand.register(this.dispatcher);
-        //WorldBorderCommand.register(this.dispatcher); // Folia - region threading - TODO later
+        WorldBorderCommand.register(this.dispatcher); // Folia - region threading - TODO later // TenseiMC - I haven't found any bugs...
         if (JvmProfiler.INSTANCE.isAvailable()) {
             JfrCommand.register(this.dispatcher);
         }
