From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Thu, 5 Oct 2023 10:40:19 +0200
Subject: [PATCH] Paper : Fix item effects conversion if no tag is present
 https://github.com/PaperMC/Paper/commit/cfe311d7a51eeb9e14c526f57efab9837f30c5d0


diff --git a/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V3568.java b/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V3568.java
index 98272d556a09e9f035005c6783ebb229c2c7157d..f84cc7fdfeba918164578ae5c74f6b697bf2dd69 100644
--- a/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V3568.java
+++ b/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V3568.java
@@ -200,6 +200,11 @@ public final class V3568 {
                 final String id = root.getString("id");
 
                 final MapType<String> tag = root.getMap("tag");
+                // TenseiMC start - Fix Paper cfe311d7a51eeb9e14c526f57efab9837f30c5d0
+                if (tag == null) {
+                    return null;
+                }
+                // TenseiMC end
 
                 if ("minecraft:suspicious_stew".equals(id)) {
                     RenameHelper.renameSingle(tag, "Effects", "effects");
